### ISSUE TITLE
Crash in fractionally scaled display in wayland (mutter)

### ISSUE THREAD

**Issue by dnut** [mullvad/mullvad-browser#132]
```
mullvad-browser 12.5.2
Mullvad MullvadBrowser 102.14.0esr
mutter 44.4
MOZ_ENABLE_WAYLAND=1
```
I have three monitors. Two outer monitors are scaled at 100% and the one in the middle is scaled at 125%. Mullvad browser usually crashes immediately when the window appears on the 125% scaled display. If I scale it down to 100%, so that all the monitors are 100% scaled, then Mullvad browser never crashes.

## Details

If I start the app on the 100% scaled display, it starts up fine. If I start the app on the 125% scaled app, sometimes it starts up fine, and sometimes it immediately crashes. Here's an example of the stdout from one of those crashes:

```
Crash Annotation GraphicsCriticalError: |[0][GFX1-]: Wayland protocol error: wl_surface@55: error 2: Buffer size (1000x985) must be an integer multiple of the buffer_scale (2).
 (t=2.1448) [GFX1-]: Wayland protocol error: wl_surface@55: error 2: Buffer size (1000x985) must be an integer multiple of the buffer_scale (2).

Gdk-Message: 22:57:57.712: Error 71 (Protocol error) dispatching to Wayland display.
Exiting due to channel error.
Crash Annotation GraphicsCriticalError: |[C0][GFX1-]: Receive IPC close with reason=AbnormalShutdown (t=0.760846)
```

Occasionally, I instead get this pop-up when starting it on the 125% scaled monitor, but usually it's just the log messages and the process exits.

![Screenshot from 2023-08-29 20-01-24](https://github.com/mullvad/mullvad-browser/assets/15116250/a3e5e9cf-b4d2-4f76-8495-58aaf7fb85fc)

If I start it on one of the 100% scaled monitors and move it to the 125% monitor, it *always* crashes as soon as it enters the new monitor, with these logs:

```
Crash Annotation GraphicsCriticalError: |[0][GFX1-]: Wayland protocol error: wl_surface@53: error 2: Buffer size (1000x985) must be an integer multiple of the buffer_scale (2).
 (t=10.089) [GFX1-]: Wayland protocol error: wl_surface@53: error 2: Buffer size (1000x985) must be an integer multiple of the buffer_scale (2).

Gdk-Message: 22:59:46.095: Error 71 (Protocol error) dispatching to Wayland display.
Exiting due to channel error.
Exiting due to channel error.
Exiting due to channel error.
Exiting due to channel error.
Exiting due to channel error.
```

If I manage to start it successfully on the 125% monitor and then move it to the 100% scaled monitor, everything is fine, until I move it back to the 125% monitor, and it *always* crashes as soon as it enters that monitor.

```
Crash Annotation GraphicsCriticalError: |[0][GFX1-]: Wayland protocol error: wl_surface@55: error 2: Buffer size (895x926) must be an integer multiple of the buffer_scale (2).
 (t=14.0405) [GFX1-]: Wayland protocol error: wl_surface@55: error 2: Buffer size (895x926) must be an integer multiple of the buffer_scale (2).

Gdk-Message: 22:59:12.233: Error 71 (Protocol error) dispatching to Wayland display.
Exiting due to channel error.
Exiting due to channel error.
Crash Annotation GraphicsCriticalError: |[C0][GFX1-]: Receive IPC close with reason=AbnormalShutdown (t=9.92389) Crash Annotation GraphicsCriticalError: |[C0][GFX1-]: Receive IPC close with reason=AbnormalShutdown (t=9.91281) Exiting due to channel error.
Crash Annotation GraphicsCriticalError: |[C0][GFX1-]: Receive IPC close with reason=AbnormalShutdown (t=11.1101) Exiting due to channel error.
Exiting due to channel error.
```

**Comment by Thorin-Oakenpants** [comment]
does it happen on the alpha version, which is based on Firefox 115 (a whole years difference from upstream)
- https://cdn.mullvad.net/browser/13.0a3/

**Comment by Thorin-Oakenpants** [comment]
woo... 
- https://bbs.archlinux.org/viewtopic.php?id=286646
- https://bugzilla.mozilla.org/show_bug.cgi?id=1834563

seems as this is solved in the MB13 (13 being the ESR115 release)

**Comment by dnut** [comment]
You're right. This problem does not exist in the latest alpha.

For arch linux users, I've published the latest alpha of Mullvad Browser in the AUR. https://aur.archlinux.org/packages/mullvad-browser-latest-bin

**Comment by ruihildt** [comment]
Fixed in Mullvad Browser 13.0.

See: https://github.com/mullvad/mullvad-browser/releases/tag/13.0
